website/api/generating-proofs/local-proving.md

```diff
@@ -6,14 +6,82 @@ The [feature flags] include options to toggle between CPU and GPU proving.
+With local proof generation your private data never leaves your machine.
 
-_With local proof generation, you can generate proofs about your private data, while your private data never leaves your machine._
+## Proving hardware
+
+RISC Zero's proving system for multiple hardware targets.
+
+:::tip
+In cases where memory is constrained (i.e. less than 10 GB is availble), it may be necessary to change the [segment size limit][segment-limit-docs-latest].
+You can find information about expected memory consumption on our [benchmarks page][datasheet].
+
+:::
+
+:::info
+The Groth16 prover currently _only_ works on x86 architecture, and so Apple Silicon is _currently unsupported_ (even via Docker).
+
+You can find out more info in the relevant issues [here](https://github.com/risc0/risc0/issues/1520) and [here](https://github.com/risc0/risc0/issues/1749).
+:::
+
+### CPU
+
+RISC Zero proving will run on nearly any modern CPU (x86 or ARM).
+Proving on CPU is the easiest to setup, and the most portable, but will not result in the highest performance or compute costs.
+
+On platforms other than MacOS, CPU proving unless otherwise configured.
+
+### NVIDIA GPU
+
+RISC Zero is highly optimized for running on GPUs. Currently, RISC Zero targets NVIDIA GPUs using the CUDA framework.
+
+Running on NVIDIA GPUS requires the CUDA runtime, and building the GPU kernels requires the NVIDIA toolkit.
+
+Using Ubuntu as an example, the following steps will install and configure the necessary dependencies, and build RISC Zero to use CUDA.
+
+```bash
+# install driver install tool
+sudo apt install ubuntu-drivers-common
+# install drivers
+sudo ubuntu-drivers install
+# install compiling tools for risc0
+sudo apt install build-essential libssl-dev -y
+# install the cuda toolkit to compile the required codebase
+sudo apt install cuda-toolkit -y
+# install rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+# setup your paths
+echo 'export PATH=/usr/local/cuda/bin:$PATH' >> ~/.bashrc
+echo 'export LD_LIBRARY_PATH=/usr/local/cuda/lib64:$LD_LIBRARY_PATH' >> ~/.bashrc
+source ~/.bashrc
+# clone the risc0 repo
+git clone https://github.com/risc0/risc0.git
+cd risc0
+# install the r0vm
+cargo install --force --path risc0/cargo-risczero
+cargo risczero install
+# run a benchmark on your GPU!
+# NOTE: Building the GPU kernels may take a long time.
+RUSTFLAGS="-C target-cpu=native" cargo run -F cuda -r --example datasheet
+```
+
+### Apple Metal
+
+On MacOS, when using a machine with Apple Silicon (such as the M-series Macbooks), RISC Zero will use the integrated [Metal][apple-metal] compute cores.
+No options need to be configured to take advantage of acceraltion through use of Metal.
+
+:::note
+When run for the first time, the Metal kernels may need to be JIT compiled.
+This can take a few minutes, but should only happen once.
```

Comment:
This applies to all GPU kernels including CUDA

Reply:
7d206eae4b66aaeba66d1b76b0fa3d51c3132b66